sources/VT100Grid.m

```diff
@@ -405,7 +405,7 @@ - (int)resetWithLineBuffer:(LineBuffer *)lineBuffer
     }
     self.cursor = VT100GridCoordMake(0, 0);
 
-    [self setCharsFrom:VT100GridCoordMake(0, preserveCursorLine ? 1 : 0)
+    [self setCharsFrom:VT100GridCoordMake(0, preservePromptLines)
```

Comment:
This will probably explode if preservePromptLines >= size_.height, so constrain it to not exceed `size_.height - 1`.